### PR TITLE
Udp supervisor butcher

### PR DESCRIPTION
**Pull request by antonbobkov**
some changes to udp server

wanted to make udp server more generic, it should know less about the game

also not sure whether it is supposed to handle signatures

### CODE REVIEW

**Comment by antonbobkov** [comment]
dialyzer fail :crying_cat_face:

**Comment by antonbobkov** [comment]
dialyzer hype :scream_cat:

File: src/p2p/potato_udp.erl [hunk 1]
```diff
@@ -3,99 +3,125 @@
...
 -type udp_address() :: {udp_ip(), udp_port()}.
+-type node_id() :: any().
+-type node_address() :: {udp_address(), node_id()}.
+-type multi_node_address() :: {udp_address(), [node_id()]}.
+
+%% maps node_id to Pid of the process
+-type node_map() :: map().
 
--type state() :: {gen_udp:socket(), map()}.
+-type state() :: {gen_udp:socket(), node_map()}.
```
Comment (pdlla) [inline src/p2p/potato_udp.erl]: :o

File: src/p2p/potato_udp.erl [hunk 2]
```diff
@@ -3,99 +3,125 @@
...
--type game_map() :: map().
 
 start_link(Port) ->
-  gen_server:start_link({local, potato_udp}, potato_udp, Port, []).
+    gen_server:start_link({local, potato_udp}, potato_udp, Port, []).
+
+-spec init(udp_port()) -> {ok, state()}.
 
-%% state maps {game_id} to game PID
--spec init(udp_port()) -> {ok, {gen_udp:socket(), game_map()}}.
```
Comment (pdlla) [inline src/p2p/potato_udp.erl]: oh, thanks for changing this to state() 😱

File: src/p2p/potato_udp.erl [hunk 3]
```diff
@@ -3,99 +3,125 @@
...
 init(Port) ->
-  %%register(potato_udp, self()),
-  io:format("gen_udp open on port: ~p~n", [Port]),
```
Comment (pdlla) [inline src/p2p/potato_udp.erl]: no io spam 😢

File: src/p2p/potato_udp.erl [hunk 4]
```diff
@@ -3,99 +3,125 @@
...
+    {ok, Socket} = gen_udp:open(Port, [binary, {active,true}]),
+    %% io:format("gen_udp open on port: ~p~n", [Port]),
+
+    {ok, {Socket, maps:new()}}.
 
 %% Never used
-handle_call(_E, _From, S) ->
-  {noreply, S}.
+handle_call(E, From, S) ->
+    erlang:error(unexptected_handle_call, [E, From, S]).
```
Comment (pdlla) [inline src/p2p/potato_udp.erl]: dialyzer doesn't break here?

File: src/p2p/potato_udp.erl [hunk 5]
```diff
@@ -3,99 +3,125 @@
...
-              end;
-            error ->
-              unknown_game_error(Packet, GameId)
-          end
-      end,
-      {noreply, S}
-  end;
+      Arg :: {add_node, node_id(), pid()}
+	     %% | {remove_node, {typetato:node_id(), typetato:verifier_id()}}
+	   | {optimized_send, multi_node_address(), any()}
```
Comment (pdlla) [inline src/p2p/potato_udp.erl]: multi_send?

File: src/p2p/potato_udp.erl (hunk 4 above)
Comment (antonbobkov) [reply src/p2p/potato_udp.erl]: :man_shrugging:

**Pull request merged**